lib/hpke/src/group.cpp

```diff
@@ -152,6 +156,91 @@ struct ECKeyGroup : public EVPGroup
   {
   }
 
+#if defined(WITH_OPENSSL3)
+  int EVP_PKEY_set_keys(EVP_PKEY* key, const bytes& sk, const bytes& pk) const
```

Comment:
This is now refactored into two functions:
- `typed_unique_ptr<EVP_PKEY> keypair_evp_key(const typed_unique_ptr<BIGNUM>& priv) const`, and
- `typed_unique_ptr<EVP_PKEY> public_evp_key(const bytes& pub) const`

I renamed the first one to use `keypair` instead of `private` to denote that the returned `EVP_PKEY` object to always contain both private and public keys.